mdm/server.go

```diff
@@ -138,8 +139,7 @@ func encodeResponse(ctx context.Context, w http.ResponseWriter, response interfa
 	}
 
 	if e, ok := response.(failer); ok && e.Failed() != nil {
-		encodeError(ctx, e.Failed(), w)
```

Comment:
I think by not calling `encodeError()` we're breaking certain MDM functionality by not returning appropriate HTTP status information from specific errors. Maybe a better option is still call `encodeError()` but have it return the error if it doesn't match the others, else nil — in this way we can both pass up the error (if need be) or handle our special MDM status codes?

Reply:
Oh, but perhaps this is what `httptransport.ServerErrorEncoder(encodeError),` does? It's already called for us? In that case, yeah, I see the issue.

Reply:
Looking at go-kit now, we also could have chosen to use StatusCoder-abiding errors to handle this logic, it seems.

Reply:
The logic in go-kit [calls encodeError](https://github.com/go-kit/kit/blob/7f14cb4dc16cd4005dc7770c131ae685cc4943b7/transport/http/server.go#L132-L134) if the response encoder returns an error.

So currently (before this PR), the response encoder calls encodeError directly and returns nil, bypassing the error logging in go-kit. Now we just return the error in the response encoder, which go-kit then logs and calls encodeError directly after.

Reply:
Resolved per our discussion during office hours.